### PR TITLE
82274 validate that payment instruction is a valid iso20022 xml

### PR DESCRIPTION
**Pull request by Topru333**

### CODE REVIEW

File: payments-workflows/src/main/kotlin/com/gevamu/corda/services/XmlService.kt [hunk 1]
```diff
@@ -37,26 +50,54 @@ open class XmlService protected constructor(
...
     protected val jaxbContext: JAXBContext = JAXBContext.newInstance(
-        *(listOf<Class<*>>(CustomerCreditTransferInitiationV09::class.java) + xmlClasses).toTypedArray()
+        *(listOf<Class<*>>(PaymentXmlData::class.java) + xmlClasses).toTypedArray()
     )
-    protected val xmlInputFactory = XMLInputFactory.newFactory()
+    protected val xmlInputFactory: XMLInputFactory = XMLInputFactory.newFactory()
+
+    private val creditTransferInitValidator: XmlValidator
+
+    private val schemaFactory by lazy { SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI) }
```
Comment (ASitnikov) [inline payments-workflows/src/main/kotlin/com/gevamu/corda/services/XmlService.kt:61]: In most cases lazy shouldn't be used in backend applications: for backend applications consistent response time is more desirable than faster start up.

File: payments-workflows/src/main/kotlin/com/gevamu/corda/services/XmlService.kt [hunk 2]
```diff
@@ -37,26 +50,54 @@ open class XmlService protected constructor(
...
+        creditTransferInitValidator = XmlValidator(
+            getCustomerCreditTransferInitiationSchema(),
+            CREDIT_TRANSFER_INIT_NAMESPACE
+        ).also {
+            it.parent = reader
+        }
+
+        val templateFactory = TransformerFactory.newInstance()
+        val xslStream = getCCTXslSchemaFile()
+        paymentTransformer = templateFactory.newTemplates(StreamSource(xslStream)).newTransformer()
```
Comment (ASitnikov) [inline payments-workflows/src/main/kotlin/com/gevamu/corda/services/XmlService.kt:76]: `Transformer` cannot be used concurrently.

**Review by ASitnikov — CHANGES_REQUESTED** [review]

File: payments-workflows/src/main/kotlin/com/gevamu/corda/iso20022/schema/XmlValidator.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        super.endPrefixMapping("")
+        super.endDocument()
+    }
+
+    override fun error(excepction: SAXParseException) {
+        throw excepction
+    }
+
+    override fun warning(excepction: SAXParseException) {
+        throw excepction
```
[the remark this answers is not in the document]
Comment (Topru333) [reply payments-workflows/src/main/kotlin/com/gevamu/corda/iso20022/schema/XmlValidator.kt:62]: will type it here  as reminder 

`
Warning Indicates that a validation event occurred that is not an error. A warning is typically issued when there is no DTD, or XML Schema to validate a particular element or attribute against. Unlike errors, warnings do not throw an exception if there is no validation event handler.
`

File: payments-workflows/src/main/kotlin/com/gevamu/corda/services/XmlService.kt [hunk 4]
```diff
@@ -37,26 +50,54 @@ open class XmlService protected constructor(
...
+
+        creditTransferInitValidator = XmlValidator(
+            getCustomerCreditTransferInitiationSchema(),
+            CREDIT_TRANSFER_INIT_NAMESPACE
+        ).also {
+            it.parent = reader
+        }
+
+        val templateFactory = TransformerFactory.newInstance()
+        val xslStream = getCCTXslSchemaFile()
```
[the remark this answers is not in the document]
Comment (Topru333) [reply payments-workflows/src/main/kotlin/com/gevamu/corda/services/XmlService.kt:75]: isnt possible in current version of kotlin for **val** values

**Review by ASitnikov — APPROVED** [review]
Looks good to me

File: payments-workflows/src/main/kotlin/com/gevamu/corda/services/XmlService.kt
Comment (ASitnikov) [inline payments-workflows/src/main/kotlin/com/gevamu/corda/services/XmlService.kt:1]: 2022-2023

File: payments-workflows/src/main/resources/com/gevamu/corda/services/pain.001.001.09.xsl [hunk 5]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright 2022-2023 Exactpro Systems Limited
```
Comment (ASitnikov) [inline payments-workflows/src/main/resources/com/gevamu/corda/services/pain.001.001.09.xsl:3]: 2022

**Review by ASitnikov — CHANGES_REQUESTED** [review]

File: payments-workflows/src/main/resources/com/gevamu/corda/services/pain.001.001.09.xsl (hunk 5 above)
Comment (Topru333) [reply payments-workflows/src/main/resources/com/gevamu/corda/services/pain.001.001.09.xsl:3]: Why 2022? 
File was created in 2022
Changes made in 2023
Commit into master will be displayed as 2023

Comment (ASitnikov) [reply payments-workflows/src/main/resources/com/gevamu/corda/services/pain.001.001.09.xsl:3]: Yes, it was a type. It should be 2023

Comment (ASitnikov) [reply payments-workflows/src/main/resources/com/gevamu/corda/services/pain.001.001.09.xsl:3]: It's a new file which is added in this MR

**Review by ASitnikov — APPROVED** [review]
Looks good to me

**Comment by fredaspen1** [comment]
Hi mates, i'm looking for dev, to a payment platform using crypto and comply to ISO20022 standard

**Comment by fredaspen1** [comment]
> Looks good to me

Hi mate, i'm looking for a dev to build a payment platform using crypto and comply with ISO20022 standard

**Pull request merged**